cupy/_math/misc.py

```diff
@@ -316,46 +316,119 @@ def clip(a, a_min=None, a_max=None, out=None):
+        'f': '__int_as_float(0xFFC00001)',
+        'd': '__longlong_as_double(0x7FF8000000000001)',
+        'F': '__int_as_float(0xFFC00001)',
+        'D': '__longlong_as_double(0x7FF8000000000001)'
+    },
+    {   # +max
+        'e': '__short_as_half(0x7BFF)',
+        'f': '__int_as_float(0x7F7FFFFF)',
+        'd': '__longlong_as_double(0x7FEFFFFFFFFFFFFF)',
+        'F': '__int_as_float(0x7F7FFFFF)',
+        'D': '__longlong_as_double(0x7FEFFFFFFFFFFFFF)'
+    },
+    {   # -max
+        'e': '__short_as_half(0xFBFF)',
+        'f': '__int_as_float(0xFF7FFFFF)',
+        'd': '__longlong_as_double(0xFFEFFFFFFFFFFFFF)',
+        'F': '__int_as_float(0xFF7FFFFF)',
+        'D': '__longlong_as_double(0xFFEFFFFFFFFFFFFF)'
+    },
+    {   # +inf
+        'e': '__short_as_half(0x7C00)',
+        'f': '__int_as_float(0x7F800000)',
+        'd': '__longlong_as_double(0x7FF0000000000000)',
+        'F': '__int_as_float(0x7F800000)',
+        'D': '__longlong_as_double(0x7FF0000000000000)'
+    },
+    {   # -inf
+        'e': '__short_as_half(0xFC00)',
+        'f': '__int_as_float(0xFF800000)',
+        'd': '__longlong_as_double(0xFFF0000000000000)',
+        'F': '__int_as_float(0xFF800000)',
+        'D': '__longlong_as_double(0xFFF0000000000000)'
+    }
+)
+
+
+def _check_nan_inf(x, default=None):
+    if x is None:
+        if default is not None:
+            return _c_nan_max_inf[default]
+        x = 0
+    if cupy.isnan(x):
+        return _c_nan_max_inf[0]
+    if cupy.isinf(x):
+        return _c_nan_max_inf[3 + (x < 0)]
+    return {
+        'e': f'static_cast<__half>({x})',
+        'f': f'static_cast<float>({x})',
+        'd': f'static_cast<double>({x})',
+        'F': f'static_cast<float>({x})',
+        'D': f'static_cast<double>({x})'
+    }
+
+
+@cupy._util.memoize()
+def _create_nan_to_num(nan, posinf, neginf):
+    nan = _check_nan_inf(nan)
+    posinf = _check_nan_inf(posinf, 1)
+    neginf = _check_nan_inf(neginf, 2)
+    return _core.create_ufunc(
+        'cupy_nan_to_num',
+        ('?->?', 'b->b', 'B->B', 'h->h', 'H->H',
+         'i->i', 'I->I', 'l->l', 'L->L', 'q->q', 'Q->Q',
+         ('e->e',
+          f'out0 = nan_to_num(in0, {nan["e"]}, {posinf["e"]}, {neginf["e"]})'),
+         ('f->f',
+          f'out0 = nan_to_num(in0, {nan["f"]}, {posinf["f"]}, {neginf["f"]})'),
+         ('d->d',
+          f'out0 = nan_to_num(in0, {nan["d"]}, {posinf["d"]}, {neginf["d"]})'),
+         ('F->F',
+          f'out0 = nan_to_num(in0, {nan["F"]}, {posinf["F"]}, {neginf["F"]})'),
+         ('D->D',
+          f'out0 = nan_to_num(in0, {nan["D"]}, {posinf["D"]}, {neginf["D"]})')),
+        'out0 = in0',
+        preamble=_nan_to_num_preamble_,
+        doc=''' Elementwise nan_to_num function.
+
+    .. seealso:: :func:`numpy.nan_to_num`
 
     ''')
 
 
+def nan_to_num(x, out=None, *, nan=0.0, posinf=None, neginf=None, **kwds):
+    """ Elementwise nan_to_num function.
+
+    .. seealso:: :func:`numpy.nan_to_num`
```

Comment:
Thanks! This also fixes the deadlink 😄 
https://docs.cupy.dev/en/latest/reference/generated/cupy.nan_to_num.html